core/src/main/scala-2/chisel3/Bits.scala

```diff
@@ -182,8 +182,6 @@ sealed abstract class Bits(private[chisel3] val width: Width) extends BitsImpl w
     * $sumWidthInt
     * @group Bitwise
     */
-  // REVIEW TODO: redundant
-  // REVIEW TODO: should these return this.type or Bits?
   final def <<(that: BigInt): Bits = macro SourceInfoWhiteboxTransform.thatArg
```

Comment:
These TODOs are actually somewhat relevant, or at least highlight something we need to handle in the Scala 3 upgrade. While `Bits.<<` returns `Bits` (even for `UInt` and `SInt`), `UInt.do_<<` returns `UInt`, and because this is a whitebox macro, it effectively means that `UInt.<<` returns `UInt` even though the ScalaDoc says it returns `UInt`.

That sounds a bit abstract, but consider the following:
```scala
class Foo extends Module {
  val in = IO(Input(UInt(8.W)))                                                                                                                                                                                 
  val out = IO(Output(UInt(8.W)))

  out :#= in >> 3 // This works because in >> 3 is a UInt
  out :#= (in: Bits) >> 3 // This doesn't because it returns Bits
}
```
Basically, we need to make sure to override `<<` (and related functions like `>>`) in our Scala 3 facades for `SInt` and `UInt` to make sure we return the right concrete subtypes.